packages/realm/src/app-services/SyncSession.ts

```diff
@@ -262,6 +263,19 @@ const CONNECTION_LISTENERS = new Listeners<ConnectionNotificationCallback, Liste
 });
 
 export class SyncSession {
+  private static instances = new IterableWeakRefs<SyncSession>();
+
+  /**
+   * Resets the internal shared pointer of all instances returned since this message was last called.
+   * @internal
+   */
+  public static resetAllInternals() {
+    for (const session of SyncSession.instances) {
+      session.resetInternal();
+    }
+    SyncSession.instances.clear();
+  }
```

Comment:
```suggestion
  public static resetAllInternals() {
    for (const session of SyncSession.instances) {
      // `resetInternal` will remove it from the collection internally
      session.resetInternal();
    }
  }
```

Reply:
Isn't that still O(N^2)? It also seems better to use `clear()` to wipe everything out in one go, rather than individually removing each entry from the set (even ignoring the fact that delete is linear). Clear is usually O(1), while deleting everything is usually O(N), but in this case is O(N^2).